model/driver/src/icon4py/model/driver/icon4py_driver.py

```diff
@@ -260,13 +247,43 @@ def _do_dyn_substepping(
             do_clean_mflx = False
 
             if not self._is_last_substep(dyn_substep):
-                self._swap()
+                prognostic_state_swp.swap()
 
             self._is_first_step_in_simulation = False
 
         # TODO (Chia Rui): compute airmass for prognostic_state here
 
 
+class DriverStates(NamedTuple):
+    """
+    Initialized states for the driver run.
+
+    Attributes:
+        prep_advection_prognostic: Fields collecting data for advection during the solve nonhydro timestep.
+        solve_nonhydro_diagnostic: Initial state for solve_nonhydro diagnostic variables.
+        diffusion_diagnostic: Initial state for diffusion diagnostic variables.
+        prognostic_swp: Initial state for prognostic variables (double buffered).
+        diagnostic: Initial state for global diagnostic variables.
+    """
+
+    prep_advection_prognostic: solve_nh_states.PrepAdvection
+    solve_nonhydro_diagnostic: solve_nh_states.DiagnosticStateNonHydro
+    diffusion_diagnostic: diffusion_states.DiffusionDiagnosticState
+    prognostic_swp: imc_utils.Swapping[prognostics.PrognosticState]
+    diagnostic: diagnostics.DiagnosticState
+
+
+class DriverParams(NamedTuple):
+    """
+    Parameters for the driver run.
+
+    Attributes:
+        divdamp_fac_o2: Second order divdamp factor.
```

Comment:
```suggestion
        divdamp_fac_o2: Second order divergence damping factor.
```
Variable name is taken directly from ICON... that is what it stands for.